### PR TITLE
RN-389: Add base map to mobile

### PR DESCRIPTION
**Pull request by edmofro**
### Issue #:
- RN-389

### Changes:
- Adds the basic map to the mobile map tab
- Can't yet select your map overlay or see a legend
- A change to the way desktop is layed out to make integration with mobile form factor more sane

### CODE REVIEW

File: packages/web-frontend/src/containers/Map/Map.js [hunk 1]
```diff
@@ -68,7 +68,8 @@ const StyledMap = styled(LeafletMap)`
  */
 class MapComponent extends Component {
   componentWillMount() {
-    window.addEventListener('resize', () => this.forceUpdate());
+    // ensure leaflet updates after 0.5 second resize animation has finished
+    window.addEventListener('resize', () => setTimeout(() => this.map?.invalidateSize(), 500));
```
Comment (edmofro) [inline packages/web-frontend/src/containers/Map/Map.js:72]: Changes in this file mainly represent the fact that we're wrapping the map in an outer div, which removes a couple of funky things that were going on:
- ZoomControl had to reposition itself based on the side panel width
- The ui-components leaflet map had to know what the side panel width was, and adjust coordinates based on that

It does add this one funky thing though as a tradeoff, which is that we need to invalidate the leaflet map after the resize animation of its wrapping container has finished

**Review by billli0 — APPROVED** [review]
LGTM! Thanks for some clean ups that makes code easier to read. 
One thing I've noticed at my end is that in desktop mode, the base map tiles are not showing. Does it occur in your end?

File: packages/web-frontend/src/screens/desktop/RootScreen/index.js [hunk 2]
```diff
@@ -47,7 +47,13 @@ const ContentContainer = styled.div`
   overflow: hidden;
 `;
 
-export const RootScreen = ({ enlargedDialogIsVisible, isLoading }) => {
+const MapContainer = styled.div`
+  height: 100vh;
+  width: calc(100vw - ${p => p.$rightOffset}px);
+  transition: width 0.5s ease;
```
[the remark this answers is not in the document]
Comment (edmofro) [reply packages/web-frontend/src/screens/desktop/RootScreen/index.js:53]: Thanks for being so thorough and testing things like this! The reason I put this in place is for when you expand and contract the sidebar - check out the zoom control for example, which jumps out suddenly if the animation isn't there
image

**Comment by edmofro** [comment]
> One thing I've noticed at my end is that in desktop mode, the base map tiles are not showing. Does it occur in your end?

Yes...It doesn't happen with the satellite option but does with open street maps. Good catch! I'll look into it.

**Comment by edmofro** [comment]
@billli0 I've been getting the blank tiles issue on the `dev` branch as well. The responses from the tile server indicate that we might be exceeding some rate limit. Strangely, when I try it on a feature instance, it doesn't have any issues (e.g. https://rn-391-mobile-map-legend.tupaia.org/ ), so maybe it's related to `localhost`? Let's see if it comes up in testing.

File: packages/web-frontend/src/screens/desktop/RootScreen/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (billli0) [reply packages/web-frontend/src/screens/desktop/RootScreen/index.js:53]: Oh that makes sense!

**Pull request merged**